<?xml version="1.0" encoding="UTF-8" ?>

<launch>
    <arg name="robot" default="X1"/>
    <arg name="world_name" default="simple_cave_03"/>
<!--    <arg name="world_name" default="finals_qual"/>-->
    <arg name="rviz" default="false"/>
    <arg name="bag" default="version-1/explorer_x1_opt_path_true_simple_cave_03_intrins_rewards_hyps_ritz_2021-08-17-15-19-24.bag"/>
<!--    <arg name="bag" default="version-1/explorer_x1_opt_path_true_ritz_finals_qual_2021-09-06-16-24-24.bag"/>-->
    <arg name="bag_path" default="$(find rpz_planning)/data/bags/mapping_eval/$(arg bag)"/>
    <arg name="bag_rate" default="4.0"/>
    <arg name="bag_start" default="0.0"/>
    <arg name="poses_csv_path" default="poses_gt.csv"/>

    <param name="use_sim_time" value="true" />

    <!-- Workaround for /tf_static problems with rosbag: https://github.com/ros/geometry2/issues/181 -->
    <node name="play_tf_static" pkg="rpz_planning" type="play_tf_static"
         args="$(arg bag_path)" output="log">
        <remap from="~tf_static" to="/tf_static"/>
    </node>

    <!-- ROSBAG -->
    <node name="rosbag_play_rpz" pkg="rosbag" type="play" output="log"
          args="--clock -s $(arg bag_start) -r $(arg bag_rate) $(arg bag_path)">
    </node>

    <group ns="$(arg robot)">
        <!-- Travelled distance publisher -->
        <include file="$(dirname)/travelled_dist.launch" pass_all_args="true">
            <arg name="route_wps_dist" value="0.5"/>
            <arg name="rate" value="1.0"/>
            <arg name="poses_gt" value="$(arg poses_csv_path)"/>
        </include>
    </group>

    <!-- RVIZ -->
    <node if="$(arg rviz)" name="rviz" pkg="rviz" type="rviz"
          args="-d $(dirname)/../config/eval.rviz"/>

</launch>
